MetasysServices/BasicServiceProvider.cs

```diff
@@ -162,12 +162,49 @@ protected List<NetworkDevice> ToNetworkDevice(List<JToken> items, ApiVersion ver
             return objects;
         }
 
+
+        /// <summary>
+        /// Gets objects from the server
+        /// </summary>
+        /// <remarks>
+        /// This method requires that Version 4 or greater of the API is being used. It overrides
+        /// the value of any <c>flatten</c> parameter to be `false` so that we get back a tree of
+        /// objects.
+        /// </remarks>
+        /// <param name="id"></param>
+        /// <param name="parameters"></param>
+        /// <returns></returns>
+        /// <exception cref="InvalidOperationException">If <see cref="Vers"/> is less than <see cref="ApiVersion.v4"/></exception>
+        protected async Task<List<MetasysObject>> GetObjectsAsync(Guid? id, Dictionary<string, string> parameters = null)

```

Comment:
The nullable `Guid` will be taken advantage of in my other PR that adds `GetObjects` without an identifier.